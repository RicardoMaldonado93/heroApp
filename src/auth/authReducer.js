import { types } from "../types/types";

export const authReducer = (state = {}, action = null) => {
  if (!action) return state;

  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true,
      };

    case types.logout:
      return {
        logged: false,
      };

    default:
      return state;
  }
};
