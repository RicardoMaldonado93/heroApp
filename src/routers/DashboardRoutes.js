import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcScreen } from '../components/dc/DcScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { HeroScreen } from '../components/ui/heroes/HeroScreen'
import { Navbar } from '../components/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container-fluid mt-2">
                <Switch>
                    <Route exact path="/marvel" component={ MarvelScreen } />
                    <Route exact path="/hero/:heroeID" component={ HeroScreen } />
                    <Route exact path="/dc" component={ DcScreen } />
                    
                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </>
    )
}
