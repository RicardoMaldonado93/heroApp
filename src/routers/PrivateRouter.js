import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types'

export const PrivateRouter = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {

    const URL = rest.location.search ? rest.location.pathname + rest.location.search : rest.location.pathname;
    localStorage.setItem("lastPath", URL );

    return (
        <Route 
            { ...rest } 
            component={ ( props ) =>(
                (isAuthenticated)
                    ? (<Component { ...props } />)
                    : (<Redirect to="/login" />)
            )}
        
        >
            
        </Route>
    )
}

PrivateRouter.propTypes ={
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}