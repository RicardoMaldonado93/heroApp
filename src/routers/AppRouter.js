import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch,
} from "react-router-dom";
import { AuthContext } from "../auth/AuthContext";
import { LoginScreen } from "../components/ui/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";
import { PrivateRouter } from "./PrivateRouter";
import { PublicRouter } from "./PublicRouter";

export const AppRouter = () => {

    const { user } = useContext(AuthContext);

    return (
        <Router>
        <div>
            <Switch>
                <PublicRouter 
                  component={ LoginScreen } 
                  exact 
                  isAuthenticated = { user.logged }
                  path="/login"
                />

                <PrivateRouter 
                  component={ DashboardRoutes } 
                  isAuthenticated = { user.logged }
                  path="/" 
                />
            </Switch>
        </div>
      </Router>
    )
}
