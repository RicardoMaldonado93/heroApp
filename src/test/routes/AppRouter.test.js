import { mount } from "enzyme";
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Test with <AppRouter />', () => {

   test("should be shown login if it isn't logged", () => {
       const contextValue = {
           dispatch: jest.fn(),
           user:{ logged: false }
       };

       const wrapper = mount(
           <AuthContext.Provider value={ contextValue }>
               <AppRouter />
           </AuthContext.Provider>
       );

       expect( wrapper ).toMatchSnapshot();
   });

   test("should be shown the marvel component if it authenticated", () => {
        const contextValue = {
            dispatch: jest.fn(),
            user:{ logged: true, name: "Ricardo" }
        };

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
 
        expect( wrapper.find(".navbar" ).exists()).toBeTruthy();
        expect( wrapper.find("h1").text().trim() ).toBe("MARVEL SCREEN");
   });
   
    
});
