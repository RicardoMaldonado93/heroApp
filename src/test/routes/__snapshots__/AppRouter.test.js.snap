// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with <AppRouter /> should be shown login if it isn't logged 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Sign In
    </button>
  </div>
</div>
`;
